opentracing/dapperish/main.go

```diff
@@ -0,0 +1,68 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"html"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"runtime"
+
+	"github.com/opentracing/api-golang/opentracing"
+)
+
+func client() {
+	reader := bufio.NewReader(os.Stdin)
+	for {
+		span, ctx := opentracing.StartSpan("getInput")
+		span.Info("ctx: ", ctx)
+		fmt.Print("Enter text: ")
+		text, _ := reader.ReadString('\n')
+		span.Info(text)
+
+		httpClient := &http.Client{}
+		httpReq, _ := http.NewRequest("POST", "http://localhost:8080/", bytes.NewReader([]byte(text)))
+		opentracing.AddContextIDToHttpHeader(span.ContextID(), httpReq.Header)
+		resp, err := httpClient.Do(httpReq)
+		if err != nil {
+			span.Error("error: ", err)
+		} else {
+			span.Info("got response: ", resp)
+		}
+
+		span.Finish()
+	}
+}
+
+func server() {
+	http.HandleFunc("/", func(w http.ResponseWriter, req *http.Request) {
+		reqCtxID, err := opentracing.GetContextIDFromHttpHeader(
+			req.Header, opentracing.GlobalTracer())
+		if err != nil {
+			panic(err)
+		}
+
+		serverSpan, _ := opentracing.StartSpan("serverSpan", reqCtxID)
```

Comment:
The `Recorder` layer is the best place to manage clock skew... and it seems utterly reasonable to use an annotation on the client and/or server side to clarify that there's an RPC between a parent and child span (at which point you can do clock skew correction in a semantically equivalent way to how Zipkin does it, even with two spans rather than one). Does that make sense?

As for the multi-hop case: this sounds like just the sort of thing that one could use span tags for... make sure that there's a clear way to identify middle-hop spans (via a well-specified tag) and infer "parentage" (causality) via something other than an explicit parent_id if those are unavailable. Happy to write more if that's not clear.